src/libcrun/container.c

```diff
@@ -3620,7 +3644,7 @@ libcrun_container_exec_with_options (libcrun_context_t *context, const char *id,
         process->apparmor_profile = xstrdup (container->container_def->process->apparmor_profile);
 
       if (process->user == NULL && container->container_def->process->user)
-        process->user = container->container_def->process->user;
+        process->user = process_user_dup (container->container_def->process->user);
```

Comment:
this function owns `container`.

Could we just steal the reference to process->user?

```
process->user = container->container_def->process->user;
container->container_def->process->user = NULL;
```

What do you think?
```

Reply:
I considered that approach, but `container` gets passed to `libcrun_join_process` and `exec_process_entrypoint` which both have a lot of complex logic, and I wasn't sure whether they would need to read the `container->container_def->process->user` (even if they aren't reading it today, they might inadvertently try to read it in the future). Let me know if you think this is not much of a concern though - happy to switch to the reference-stealing approach if you think it's OK.

Reply:
you are right, better to not mess with the fields as it is difficult to track what would happen if we reset it.

I can't think of any better way (except changing libocispec to generate "clone" operations, but that is too much work for such a simple fix), so I'll just merge the current version